src/ObjCRuntime/RuntimeOptions.cs

```diff
@@ -193,20 +193,28 @@ internal static TypeDefinition GetHttpMessageHandler (Application app, RuntimeOp
 			return type;
 		}
 #else
+
+#if SYSTEM_NET_HTTP
 		[Register ("__internal__xamarin_runtimeoptions_bundlefinder")]
 		class BundleFinder : NSObject {
 			public BundleFinder ()
 			{
 				IsDirectBinding = false;
 			}
 		}
+#endif
 
 		internal static RuntimeOptions Read ()
 		{
 			// for iOS NSBundle.ResourcePath returns the path to the root of the app bundle
 			// for macOS apps NSBundle.ResourcePath returns foo.app/Contents/Resources
 			// for macOS frameworks NSBundle.ResourcePath returns foo.app/Versions/Current/Resources
-			var resource_dir = NSBundle.FromClass (new Class (typeof (BundleFinder))).ResourcePath;
+#if SYSTEM_NET_HTTP
+			Class bundle_finder = new Class (typeof (BundleFinder));
+#else
+			Class bundle_finder = new Class (typeof (NSObject.NSObject_Disposer));
+#endif
```

Comment:
@spouliot I've now added an `[InternalsVisibleTo]` from Xamarin.*.dll to System.Net.Http.dll, which simplified the code considerably :smile: